test/jest/unit/lib/commands/fix/fix.spec.ts

```diff
@@ -294,6 +298,93 @@ describe('snyk fix (functional tests)', () => {
     testTimeout,
   );
 
+  it(
+    'snyk fix send errors as analytics when fix fails',
+    async () => {
+      // read data from console.log
+      let stdoutMessages = '';
+      let stderrMessages = '';
+      jest
+        .spyOn(console, 'log')
+        .mockImplementation((msg: string) => (stdoutMessages += msg));
+      jest
+        .spyOn(console, 'error')
+        .mockImplementation((msg: string) => (stderrMessages += msg));
+      jest.spyOn(snyk, 'test').mockResolvedValue({
+        ...pipWithRemediation,
+        // pip plugin does not return targetFile, instead fix will fallback to displayTargetFile
+        displayTargetFile: pipRequirementsTxt,
+      });
+      snykFixSpy.mockResolvedValueOnce({
+        results: {
+          python: {
+            failed: [
+              {
+                original: [],
+                error: new Error(
+                  'SolverProblemError Because x (2.6) depends on y (>=1.19)',
+                ),
+                tip: 'Try running `pipenv install x==2.6 transitive==1.1.1',
+              },
+            ],
+            skipped: [],
+            succeeded: [],
+          },
+        },
+        exceptions: {},
+        meta: {
+          fixed: 0,
+          failed: 1,
+          fixableIssues: 14,
+          fixedIssues: 0,
+          totalIssues: 14,
+        },
+        fixSummary: '✖ No successful fixes',
+      });
+
+      let res;
+      try {
+        res = await fix(pipAppWorkspace, {
+          dryRun: true,
+          quiet: true,
+          _doubleDashArgs: [],
+          _: [],
+        });
+      } catch (e) {
+        res = e.message;
+      }
+      expect(snykFixSpy.mock.calls[0][1]).toEqual({
+        dryRun: true,
+        quiet: true,
+      });
```

Comment:
i find this a little more readable

```suggestion
      expect(snykFixSpy).toHaveBeenCalledWith({
        dryRun: true,
        quiet: true,
      });
```

Reply:
unfortunately if I do that it expects me to pass in the entirety of the options sent and it's a large array of object from CLI